### ISSUE TITLE
Android 10

### ISSUE THREAD

**Issue by ghost** [Lanchon/haystack#31]

**Comment by jpourmalek** [comment]
we are waiting for your update for android 10 dude

**Comment by Lanchon** [comment]
i am not working on it. all roms seem to be odexed.

**Comment by waltercool** [comment]
@Lanchon  Android 10 can be compiled, even LineageOS is available at their 17th version.

How can I help to make this a solution? I tried this patch, and might work https://github.com/microg/android_packages_apps_GmsCore/pull/957

**Comment by ghost** [comment]
also it appears that some of the jars that haystack patches are in APEX packages now.

**Comment by Lanchon** [comment]
@steamp0rt,

yes i know, though android rom builds default to flattened apex mode which means the plain old files are accessible somewhere by default.

and odex files inside apex files can possibly be in cdex format.

the open source community should really fork away from google's android once and for all. we should all be seen externally as android 7 maybe. and devs who want to stop support of android 7 loose a sizable fraction of power users. if we are enough, they won't do it.

**Comment by Nanolx** [comment]
APEX isn't so much of a problem, you can easily mount those files loop-back, they look like normal odex files inside and vdexextractor was updated with Android 10 support just recently.

The code I use for mounting APEX for proper working environment in TWRP can be found here: https://gitlab.com/Nanolx/NanoDroid/blob/master/CommonInstaller#L231

**Comment by anoosa1** [comment]
any updates?
